File: app/src/main/java/com/worldventures/dreamtrips/modules/membership/bundle/UrlBundle.java
package com.worldventures.dreamtrips.modules.membership.bundle;

import android.os.Parcel;
import android.os.Parcelable;

public class UrlBundle implements Parcelable {

   private String url;

   public UrlBundle(String url) {
      this.url = url;
   }

   protected UrlBundle(Parcel in) {
      url = in.readString();
   }

   public static final Creator<UrlBundle> CREATOR = new Creator<UrlBundle>() {
      @Override
      public UrlBundle createFromParcel(Parcel in) {
         return new UrlBundle(in);
      }

      @Override
      public UrlBundle[] newArray(int size) {
         return new UrlBundle[size];
      }
   };

   public String getUrl() {
      return url;
   }

   @Override
   public void writeToParcel(Parcel parcel, int i) {

      parcel.writeString(url);
   }

   @Override
   public int describeContents() {
      return 0;
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/social/ui/membership/bundle/UrlBundle.kt
package com.worldventures.dreamtrips.social.ui.membership.bundle

import android.os.Parcel
import android.os.Parcelable

data class UrlBundle(val url: String?) : Parcelable {

   constructor(source: Parcel) : this(
         source.readString()
   )

   override fun describeContents() = 0

   override fun writeToParcel(dest: Parcel, flags: Int) = with(dest) {
      writeString(url)
   }

   companion object {
      @JvmField
      val CREATOR: Parcelable.Creator<UrlBundle> = object : Parcelable.Creator<UrlBundle> {
         override fun createFromParcel(source: Parcel): UrlBundle = UrlBundle(source)
         override fun newArray(size: Int): Array<UrlBundle?> = arrayOfNulls(size)
      }
   }
}
